cpp/lib/ome/common/units/pressure.h

```diff
@@ -0,0 +1,403 @@
+      /// Measured quantity in kilobars.
+      typedef quantity<kilobar_unit> kilobar_quantity;
+      /// Measured quantity in megabars.
+      typedef quantity<megabar_unit> megabar_quantity;
+
+      /// Unit definition for atmosphere pressure.
+      typedef boost::units::metric::atmosphere_base_unit::unit_type atmosphere_unit;
+      /// Numeric constant for atmosphere.
+      BOOST_UNITS_STATIC_CONSTANT(atmosphere, atmosphere_unit);
+      /// Numeric constant for atmosphere.
+      BOOST_UNITS_STATIC_CONSTANT(atmospheres, atmosphere_unit);
+      /// Measured quantity in atmospheres.
+      typedef quantity<atmosphere_unit> atmosphere_quantity;
+
+      /// Base unit for pound-force per square inch (psi) pressure.
+      struct psi_base_unit : boost::units::base_unit<psi_base_unit,
+                                                     boost::units::pressure_dimension, 3> { };
+      /// Unit definition for pound-force per square inch (psi) pressure.
+      typedef psi_base_unit::unit_type psi_unit;
+      /// Numeric constant for pound-force per square inch (psi).
+      BOOST_UNITS_STATIC_CONSTANT(psi, psi_unit);
+      /// Measured quantity in pound-force per square inch (psi).
+      typedef quantity<psi_unit> psi_quantity;
+
+      /// Unit definition for Torr pressure.
+      typedef boost::units::metric::torr_base_unit::unit_type torr_unit;
+      /// Numeric constant for Torr.
+      BOOST_UNITS_STATIC_CONSTANT(torr, torr_unit);
+      /// Measured quantity in Torrs.
+      typedef quantity<torr_unit> torr_quantity;
+
+      /// Unit definition for milliTorr pressure.
+      typedef scaled_base_unit<boost::units::metric::torr_base_unit,scale<10,static_rational< -3> > >::unit_type millitorr_unit;
+      /// Numeric constant for milliTorr.
+      BOOST_UNITS_STATIC_CONSTANT(millitorr, millitorr_unit);
+      /// Measured quantity in milliTorrs.
+      typedef quantity<millitorr_unit> millitorr_quantity;
+
+      /// Unit definition for mmHg pressure.
+      typedef boost::units::metric::mmHg_base_unit::unit_type mmHg_unit;
+      /// Numeric constant for mmHg.
+      BOOST_UNITS_STATIC_CONSTANT(mmHg, mmHg_unit);
+      /// Measured quantity in mmHg.
+      typedef quantity<mmHg_unit> mmHg_quantity;
+
+    }
+  }
+}
+
+/// Conversion factor for pound-force per square inch (psi) to pascal.
+BOOST_UNITS_DEFINE_CONVERSION_FACTOR(ome::common::units::psi_base_unit, si::pressure, double, 6894.7573); // exact conversion
+/// Default conversion for pound-force per square inch (psi) to is to SI pressure units (pascals).
+BOOST_UNITS_DEFAULT_CONVERSION(ome::common::units::psi_base_unit, si::pressure);
+
+namespace boost
+{
+  namespace units
+  {
+
+    /// Unit information for pound-force per square inch (psi).
+    template<> struct base_unit_info<ome::common::units::psi_base_unit>
+    {
+      /// Unit name.
+      static std::string name()   { return "pound-force per square inch"; }
```

Comment:
They will do, in the model units code.  But not at this level--this is the low-level unit conversion; the Java code doesn't do this.  This will mostly be hidden entirely unless you need to do explicit unit conversion; the following PR should make this a bit clearer; see my `xsdfu-cpp-2015` branch (incomplete work in progress https://github.com/rleigh-dundee/bioformats/tree/xsdfu-cpp-2015).

However, note I found some fundamental problems with the Java units code which weren't picked up in the code review at the time; I certainly don't want to blindly copy something which is defective.  If we could have a chat about this that would be great.